Lib/test/test_import/__init__.py

```diff
@@ -807,6 +807,14 @@ def test_issue105979(self):
         self.assertIn("Frozen object named 'x' is invalid",
                       str(cm.exception))
 
+    def test_frozen_module_from_import_error(self):
+        with self.assertRaises(ImportError) as cm:
+            from os import this_will_never_exist
+        self.assertRegex(
+            str(cm.exception),
+            r"cannot import name 'this_will_never_exist' from 'os' \(.*Lib[\\/]os\.py\)"
```

Comment:
Is it the right path? What if Python is installed or the stdlib is precompiled or packed in the ZIP archive? Maybe use `os.__file__`?

Reply:
Thank you, fixed